d3-series/materials/Week 4/Assignment/js/air_quality_chart.js

```diff
@@ -0,0 +1,147 @@
+function buildChart(containerId) {
+    // read in our data
+    d3.csv('data/air_quality.csv', function(error, data) {
+        // handle read errors
+        if (error) {
+            console.error('failed to read data');
+            return;
+        }
+
+        console.log('raw', data);
+
+        // coerce data to numeric 
+        data.forEach(function(d) {
+            d.Emissions = parseFloat(d.Emissions.replace(/,/g, ''));
+        });
+
+        console.log('clean', data);
+ 
+    // size globals
+    var margin = {
+        top: 50,
+        right: 80,
+        bottom: 50,
+        left: 80
+    };
+
+    // calculate dimensions
+    var width = 1300;
+    var height = 500;
+    var innerWidth = width - margin.left - margin.right;
+    var innerHeight = height - margin.top - margin.bottom;
+
+    // create svg element
+    var svg = d3
+        .select(containerId)
+        .append('svg')
+        .attr('height', height)
+        .attr('width', width);
+
+    // create inner group element
+    var g = svg
+        .append('g')
+        .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+
+        // scales
```

Comment:
Not sure what editor you are using to work on, but on vs code there is an extension called Prettier that will automatically format the code for you